cnf-certification-test/chaostesting/suite.go

```diff
@@ -0,0 +1,280 @@
+	}
+	gvr := schema.GroupVersionResource{Group: "litmuschaos.io", Version: "v1alpha1", Resource: "chaosengines"}
+	if err := oc.DynamicClient.Resource(gvr).Namespace(namespace).Delete(context.TODO(), "engine-test", deleteOptions); err != nil {
+		logrus.Debugf("error while removing the chaos engine resources %e", err)
+	}
+	err := oc.K8sClient.CoreV1().ServiceAccounts(namespace).Delete(context.TODO(), "test-sa", deleteOptions)
+	if err != nil {
+		logrus.Debugf("error while removing the ServiceAccountsresources %e", err)
+	}
+	if err = oc.RbacClient.Roles(namespace).Delete(context.TODO(), "test-sa", deleteOptions); err != nil {
+		logrus.Debugf("error while removing the chaos engine resources %e", err)
+	}
+	if err = oc.RbacClient.RoleBindings(namespace).Delete(context.TODO(), "test-sa", deleteOptions); err != nil {
+		logrus.Debugf("error while removing the chaos engine resources %e", err)
+	}
+	gvr = schema.GroupVersionResource{Group: "litmuschaos.io", Version: "v1alpha1", Resource: "chaosexperiments"}
+	if err := oc.DynamicClient.Resource(gvr).Namespace(namespace).Delete(context.TODO(), chaosname, deleteOptions); err != nil {
+		logrus.Debugf("error while removing the chaos engine resources %e", err)
+	}
+	e := os.Remove("chaostesting/chaosEngine.yaml.tmp")
+	if e != nil {
+		logrus.Debugf("error while removing the temp file of the chaos engine %e", e)
+	}
+	e = os.Remove("chaostesting/servicAccount.yaml.tmp")
+	if e != nil {
+		logrus.Debugf("error while removing the temp file of the servicAccount %e", e)
+	}
+	e = os.Remove("chaostesting/deleteExperment.yaml.tmp")
+	if e != nil {
+		logrus.Debugf("error while removing the temp file of the deleteExperment %e", e)
+	}
+}
+
+func applyTemplate(appLabel, appKind, namespace, fileename string) (string, error) {
+	input, err := os.ReadFile(fileename)
+	if err != nil {
+		fmt.Println(err)
+		return "", err
+	}
+	output := bytes.ReplaceAll(input, []byte("{{ APP_NAMESPACE }}"), []byte(namespace))
+	output = bytes.ReplaceAll(output, []byte("{{ APP_LABEL }}"), []byte(appLabel))
+	output = bytes.ReplaceAll(output, []byte("{{ APP_KIND }}"), []byte(appKind))
+	fileName := fileename + ".tmp"
+	if err = os.WriteFile(fileName, output, 0700); err != nil {
+		fmt.Println(err)
+		return "", err
+	}
+	return fileName, nil
+}
+
+func waitForTestFinish(timeout time.Duration) bool {
+	const pollingPeriod = 1 * time.Second
+	var elapsed time.Duration
+	var result bool
+	for elapsed < timeout {
+		result = waitForResult()
+
+		if result {
+			break
+		}
+		time.Sleep(pollingPeriod)
+		elapsed += pollingPeriod
+	}
+	return result
+}
+
+func returnResult() (bool, string) {
+	oc := clientsholder.GetClientsHolder()
+	gvr := schema.GroupVersionResource{Group: "litmuschaos.io", Version: "v1alpha1", Resource: "chaosresults"}
+	crs, err := oc.DynamicClient.Resource(gvr).List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		logrus.Errorf("error getting : %v\n", err)
+	}
+	for _, cr := range crs.Items {
```

Comment:
If you're not expecting more than one chaos experiment result, you should check `len(crs.Items) == 1` here.